bin/useLayoutEffectRule.js

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+module.exports = function (context) {
+  return {
+    ImportDeclaration(node) {
+      const source = node.source.value;
+      if (source === '@react-aria/utils' || source === './useLayoutEffect' || source === './') {
+        return;
+      }
+      const importSpecifiers = node.specifiers.filter(specifier => specifier.type === 'ImportSpecifier');
+      const getName = specifier => specifier.local.name;
+      importSpecifiers.map(
+                (item) => {
+                  let itemName = getName(item);
+                  if (itemName === 'useLayoutEffect') {
+                    context.report(node, 'Please use useLayoutEffect from @react-aria/utils instead.');
+                  }
+                });
+      return;
```

Comment:
Small formatting nit:
```suggestion
      importSpecifiers.map(
        (item) => {
          let itemName = getName(item);
          if (itemName === 'useLayoutEffect') {
            context.report(node, 'Please use useLayoutEffect from @react-aria/utils instead.');
          }
        }
      );
```